### PR TITLE
fix: ACL user can't delete media files

### PR DESCRIPTION
**Pull request by chinnu-focaloid**
Relates to https://oat-sa.atlassian.net/browse/ADF-489
UDIR - "ACL" User can't delete media files in a item

### CODE REVIEW

File: actions/class.ItemContent.php [hunk 1]
```diff
@@ -202,7 +202,7 @@ private function getMediaAsset(bool $validateWriteAccess = false): ?MediaAsset
     {
         $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
         $asset = $this->resolveAsset($params['uri'], $params['path'], $params['lang']);
-        $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+        $resourceUri = $params['uri'];
```
Comment (gabrielfs7) [inline actions/class.ItemContent.php:205]: We need to make sure it wont break ACL for assets (not exclusive item files). It should support both ways.

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: actions/class.ItemContent.php (hunk 1 above)
Comment (chinnu-focaloid) [reply actions/class.ItemContent.php:205]: It is working fine for assets too.

Comment (gabrielfs7) [reply actions/class.ItemContent.php:205]: And for assets, will the `$params['uri']` be the item URI or the asset URI?

Comment (chinnu-focaloid) [reply actions/class.ItemContent.php:205]: It is Item URI

Comment (gabrielfs7) [reply actions/class.ItemContent.php:205]: So, that means that we do not check permissions for the asset properly.

File: actions/class.ItemContent.php [hunk 2]
```diff
@@ -34,6 +34,7 @@
 use oat\taoItems\model\media\AssetTreeBuilder;
 use oat\taoItems\model\media\AssetTreeBuilderInterface;
 use oat\taoItems\model\media\ItemMediaResolver;
+use oat\taoMediaManager\model\MediaSource;
```
Comment (gabrielfs7) [inline actions/class.ItemContent.php:37]: You cannot add this here, cause it is part of taoMediaManager and this extension is not coupled to it.

File: actions/class.ItemContent.php [hunk 3]
```diff
@@ -202,7 +203,13 @@ private function getMediaAsset(bool $validateWriteAccess = false): ?MediaAsset
     {
         $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
         $asset = $this->resolveAsset($params['uri'], $params['path'], $params['lang']);
-        $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+
+        $mediaSource = $asset->getMediaSource();
+        if ($mediaSource instanceof MediaSource) {
```
Comment (gabrielfs7) [inline actions/class.ItemContent.php:208]: You have some options here, maybe: 

- Check based on the path you received.
- Check if they are instance of `HttpSource` or `LocalItemSource` instead.

File: actions/class.ItemContent.php [hunk 4]
```diff
@@ -202,7 +203,13 @@ private function getMediaAsset(bool $validateWriteAccess = false): ?MediaAsset
...
         $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
         $asset = $this->resolveAsset($params['uri'], $params['path'], $params['lang']);
-        $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+
+        $mediaSource = $asset->getMediaSource();
+        if ($mediaSource instanceof LocalItemSource) {
+            $resourceUri = $params['uri'];
+        } else {
+            $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+        }
```
Comment (gabrielfs7) [inline actions/class.ItemContent.php:212]: We can simplify the code:

```suggestion
        $resourceUri = $asset->getMediaSource() instanceof LocalItemSource) 
            ? $params['uri']
            : tao_helpers_Uri::decode($asset->getMediaIdentifier());
```

Comment (chinnu-focaloid) [reply actions/class.ItemContent.php]: Done

**Review by gabrielfs7 — APPROVED** [review]
Good job! Please, prepare a demo and also test on UDIR demo @chinnu-focaloid

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on https://demo.udir.taocloud.org/

**Pull request merged**